arroyo-worker/src/connectors/kafka/sink/mod.rs

```diff
@@ -150,4 +236,49 @@ impl<K: Key + Serialize, T: Data + Serialize> KafkaSinkFunc<K, T> {
 
         self.publish(k, v).await;
     }
+
+    async fn handle_commit(&mut self, epoch: u32, ctx: &mut crate::engine::Context<(), ()>) {
+        let ConsistencyMode::ExactlyOnce { next_transaction_index: _, producer_to_complete } = &mut self.consistency_mode else {
+            warn!("received commit but consistency mode is not exactly once");
+            return
+        };
+
+        let Some(committing_producer)= producer_to_complete.take() else {
+            unimplemented!("received a commit message without a producer ready to commit. Restoring from commit phase not yet implemented");
+        };
+        let mut commits_attempted = 0;
+        while commits_attempted < 5 {
+            if committing_producer
+                .commit_transaction(Timeout::After(Duration::from_secs(10)))
+                .is_ok()
+            {
+                break;
+            } else {
+                error!("failed to commit {} times, retrying", commits_attempted);
+                commits_attempted += 1;
+            }
+        }
```

Comment:
Should this panic if we fail all 5 attempts?

Reply:
doh